Frontend/Pages/Projects/Features/Reason.cshtml.cs

```diff
@@ -13,60 +16,65 @@ public class Reason : CommonPageModel
     {
         private readonly IProjects _projects;
 
-        [BindProperty] public FeaturesReasonViewModel FeaturesReasonViewModel { get; set; } = new FeaturesReasonViewModel();
-
         public Reason(IProjects projects)
         {
             _projects = projects;
         }
 
+        [BindProperty]
+        public TransferFeatures.ReasonForTheTransferTypes ReasonForTheTransfer { get; set; }
+
         public async Task<IActionResult> OnGetAsync()
         {
-
             var project = await _projects.GetByUrn(Urn);
-            
+
             var projectResult = project.Result;
             IncomingTrustName = projectResult.IncomingTrustName;
-            FeaturesReasonViewModel.IsSubjectToIntervention =
-                projectResult.Features.ReasonForTransfer.IsSubjectToRddOrEsfaIntervention;
-            FeaturesReasonViewModel.MoreDetail = projectResult.Features.ReasonForTransfer.InterventionDetails;
-
+            ReasonForTheTransfer = projectResult.Features.ReasonForTheTransfer;
             return Page();
         }
 
-        public List<RadioButtonViewModel> ReasonRadioButtons()
-        {
-            var result = new[] {true, false}.Select(value => new RadioButtonViewModel
-            {
-                Value = value.ToString(),
-                Name = nameof(FeaturesReasonViewModel.IsSubjectToIntervention),
-                DisplayName = value ? "Yes" : "No",
-                Checked = FeaturesReasonViewModel.IsSubjectToIntervention == value
-            }).ToList();
-
-            return result;
-        }
-
         public async Task<IActionResult> OnPostAsync()
         {
             var project = await _projects.GetByUrn(Urn);
-            
+
+            var validator = new FeaturesReasonValidator();
+            var validationResults = await validator.ValidateAsync(this);
```

Comment:
as you say this can be done on binding if it is in a seperate class, but no problem with this also